desktop/core/src/desktop/js/parse/sql/generic/jison/select/select.test.json

```diff
@@ -3927,6 +3927,70 @@
       "lowerCase": false
     }
   },
+  {
+    "namePrefix": "should suggest keywords",
+    "beforeCursor": "select bla::",
+    "afterCursor": "",
+    "containsKeywords": [
+      "INT",
+      "STRING"
+    ],
+    "expectedResult": {
+      "lowerCase": true
+    }
+  },
+  {
+    "namePrefix": "should suggest keywords",
+    "beforeCursor": "SELECT bla::ST",
+    "afterCursor": " FROM bar;",
+    "containsKeywords": [
+      "INT",
+      "STRING"
+    ],
+    "expectedResult": {
+      "lowerCase": false
+    }
+  },  
+  {
+    "namePrefix": "should suggest keywords",
+    "beforeCursor": "SELECT (5 + 5)::",
+    "afterCursor": " FROM bar;",
+    "containsKeywords": [
+      "INT",
+      "STRING"
+    ],
+    "expectedResult": {
+      "lowerCase": false
+    }
+  },
+  {
+    "namePrefix": "should suggest columns for order by after a cast in a join",
+    "beforeCursor": "select * from a join b on a.a = b.b::date order by ",
```

Comment:
These use to not work at all and now works, which is a nice win. 
However, if your casted type doesn't match one of the above types, then things still don't work. I wish the below would still work correctly also, not sure if there's a good way to accomplish that or not:

```  
select * from a join b on a.a = b.b::dte order by ",
```